eox_core/social_tpa_backends.py

```diff
@@ -61,3 +63,44 @@ def extra_data(self, user, uid, response, details=None, *args, **kwargs):  # pyl
         data["user_details"] = self.get_user_details(response)
 
         return data
+
+    def get_user_id(self, *args, **kwargs):
+        """
+        Insert a slug at the beginning of the user id. Retroactively add the
+        slug for social users that don't have one. The slug must be defined
+        in the 'OTHER_SETTINGS' field from the provider configuration.
+        In case a slug was not defined raise an exception due to misconfigured
+        backend.
+        This three behaviours can be controlled via the settings:
+            - 'SOCIAL_AUTH_NAMESPACED_UIDS': if false behave as originally
+            intended. If true, add a slug to the uid.
+            - 'SOCIAL_AUTH_ALLOW_SLUGLESS_UID': if false raise and exception when no slug
+            is defined. If true, return de original uid when no slug is defined.
+            - 'SOCIAL_AUTH_ALLOW_WRITE_SLUG_UID': If true update the uid of existing social
+            users if any.
```

Comment:
Could you please add an example here of what the settings json should have for this to:1) convert old style into new and 2) work only when the new format is updated in all the socialauths

Reply:
Sure thing.